src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java

```diff
@@ -23,8 +45,76 @@ public RewardScoreEntity recalculateScore(AllRewardScoresEntity allRewardScores,
     public RewardScoreEntity calculateOnContentWorkedOn(AllRewardScoresEntity allRewardScores, List<Content> contents, UserProgressLogEvent event) {
         RewardScoreEntity fitnessScoreBefore = allRewardScores.getFitness();
 
-        // TODO calculate fitness score
+        double fitnessDecrease = calculateFitnessDecrease(contents, event);
+        double newFitnessScore = fitnessScoreBefore.getValue() * (1 - fitnessDecrease);
+        int newFitnessScoreInt = (int) Math.round(newFitnessScore);
+
+        // Decrease the fitness score
+        int fitnessDecreaseValue = fitnessScoreBefore.getValue() - newFitnessScoreInt;
+        int updatedFitnessScore = fitnessScoreBefore.getValue() - fitnessDecreaseValue;
+
+        RewardScoreLogEntry logEntry = RewardScoreLogEntry.builder()
+                .date(OffsetDateTime.now())
+                .difference(fitnessDecreaseValue)
+                .oldValue(fitnessScoreBefore.getValue())
+                .newValue(updatedFitnessScore)
+                .reason(RewardChangeReason.CONTENT_DONE)
+                .associatedContentIds(Collections.emptyList())
```

Comment:
This should not be empty, the user should know which content it was, that increased his fitness